hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/opt/physical/CreateDagVisitor.java

```diff
@@ -126,10 +134,32 @@ public Vertex onProject(ProjectPhysicalRel rel) {
                 (BiFunctionEx<Function<Object[], Object[]>, Object[], Object[]>) Function::apply
         ));
 
-        connectInput(rel.getInput(), vertex, null);
+        connectInputPreserveCollation(rel, vertex);
         return vertex;
     }
 
+    public Vertex onSort(SortPhysicalRel rel) {
```

Comment:
Could you please, describe an algorithm in comment? It would be extremely great for fast understanding.